pkg/apis/core/types.go

```diff
@@ -3960,7 +3960,7 @@ type AppArmorProfile struct {
 	//   Localhost - a profile pre-loaded on the node.
 	//   RuntimeDefault - the container runtime's default profile.
 	//   Unconfined - no AppArmor enforcement.
-	// +unionDescriminator
+	// +unionDiscriminator
 	Type AppArmorProfileType
```

Comment:
It looks like the [external types](https://github.com/kubernetes/kubernetes/blob/68899f8e6d5861e7b6197c51b0dee9f8a486e3e0/staging/src/k8s.io/api/core/v1/types.go#L4569-L4570) have this value correct, and the [validation](https://github.com/kubernetes/kubernetes/blob/68899f8e6d5861e7b6197c51b0dee9f8a486e3e0/pkg/apis/core/validation/validation.go#L4833-L4873) is ensuring we only hold correct values.

Reply:
unqualified tags (+optional, +required, +default, +union, +unionDiscriminator)
k8s tags (+k8s:...)
other tags (+kubebuilder:...)

Maybe we could add warning logic in kube-openapi that _at least_ warns if an unrecognized unqualified or `k8s:...` tag is used

Reply:
@soltysh can you open issue with ^ in kube-openapi repo to track?

Reply:
@liggitt @soltysh Thank you for the review. I opened an issue: https://github.com/kubernetes/kube-openapi/issues/525